lib/Controller/DelegationController.php

```diff
@@ -54,10 +75,163 @@ public function getAllGroups(): DataResponse {
 			$data[] = [
 				'id' => $group->getGID(),
 				'displayname' => $group->getDisplayName(),
+				'usercount' => $group->count(),
+				'disabled' => $group->countDisabled(),
+				'canAdd' => $group->canAddUser(),
+				'canRemove' => $group->canRemoveUser(),
 			];
 		}
 
 		// return info
 		return new DataResponse($data);
 	}
+
+	/**
+	 * Get the list of groups allowed to use groupfolders
+	 *
+	 * @NoAdminRequired
+	 * @RequireGroupFolderAdmin
+	 *
+	 * @return DataResponse
+	 */
+	public function getAllowedGroups() {
+		$groups = json_decode($this->config->getAppValue('groupfolders', 'delegated-admins', '[]'));
+
+		// transform in a format suitable for the app
+		$data = [];
+		foreach ($groups as $gid) {
+			$group = $this->groupManager->get($gid);
+			$data[] = [
+				'id' => $group->getGID(),
+				'displayname' => $group->getDisplayName(),
+				'usercount' => $group->count(),
+				'disabled' => $group->countDisabled(),
+				'canAdd' => $group->canAddUser(),
+				'canRemove' => $group->canRemoveUser(),
+			];
+		}
+		return new DataResponse($data);
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @return DataResponse
+	 */
+	public function getAuthorizedGroups() {
```

Comment:
```suggestion
	public function getAuthorizedGroups(): DataResponse {
```

and then you don't need the phpdoc, same can be done in other places

Reply:
Should I delete the `@return DataResponse` php docs ?